hidden-posts.php

```diff
@@ -128,6 +116,36 @@ static function remove_post( $id ) {
         update_option( self::META_KEY, array_map( 'intval', $posts ) );
     }
 
+    /**
+     * Render the meta box for hiding a post.
+     */
+    public function add_metabox() {
+        add_meta_box(
+            'hidden-posts',
+            esc_html( apply_filters( 'hidden_posts_checkbox_title', 'Visibility' ) ),
+            array( $this, 'render_metabox' ),
+            'post',
+            'side',
+            'high'
+        );
+    }
+
+    /**
+     * Render the meta box for hiding a post.
+     *
+     * @param WP_Post $post The post object for which the metabox should be added.
+     */
+    public function render_metabox( WP_Post $post ) {
+        $checked = in_array( (int) $post->ID, self::get_posts(), true );
```

Comment:
Though I don't see any error in my log but I wonder how $post is passed here. 
If it's a global $post, it should be declared like the original code: `global $post` before it can be used.

Reply:
`$post` comes from the callback function in line 126. This callback function expects the `$post` variable as default value. 

Does that answer your question?